pybossa/view/projects.py

```diff
@@ -2171,7 +2317,11 @@ def task_timeout(short_name):
                                                                     current_user,
                                                                     ps)
     title = project_title(project, gettext('Timeout'))
-    form = TaskTimeoutForm()
+    if request.headers.get('content-type') == 'application/json' and request.data:
```

Comment:
Is there a reason why you need to do this at each handler's level and the [`before_request` callback](https://github.com/bloomberg/pybossa/blob/master/pybossa/core.py#L580) does not work?

Reply:
will fix it in another PR